File: scripts/convert-images.js
import fs from 'fs';
import path from 'path';
import sharp from 'sharp';

const directoryPath = path.join(process.cwd(), 'src/frontend/public/assets/generated');

async function convertToWebP() {
  try {
    const files = fs.readdirSync(directoryPath);

    for (const file of files) {
      if (file.endsWith('.jpg') || file.endsWith('.png')) {
        const filePath = path.join(directoryPath, file);
        // We will keep the original file name structure and just change extension to .webp
        // E.g., hero-jewellery-banner.dim_1600x700.jpg -> hero-jewellery-banner.dim_1600x700.webp
        const outputFilePath = path.join(directoryPath, file.replace(/\.(jpg|png)$/, '.webp'));

        console.log(`Converting ${file} to WebP...`);
        
        await sharp(filePath)
          .webp({ quality: 80 })
          .toFile(outputFilePath);
          
        console.log(`Successfully converted ${file}`);
        
        // Optionally, delete the old file to save space and ensure we use the new one
        // fs.unlinkSync(filePath);
      }
    }
    console.log('All images converted successfully!');
  } catch (err) {
    console.error('Error converting images:', err);
  }
}

convertToWebP();

File: convert_images.py
import os
from PIL import Image

folder_path = r'd:\web_code\gemora-global\src\frontend\public\assets\generated'

for filename in os.listdir(folder_path):
    if filename.lower().endswith(('.jpg', '.png')):
        file_path = os.path.join(folder_path, filename)
        name, ext = os.path.splitext(filename)
        output_path = os.path.join(folder_path, name + '.webp')
        
        try:
            with Image.open(file_path) as img:
                # Convert RGBA to RGB for JPEG-like WebP, but WebP supports transparency.
                # Just save it directly as WebP
                img.save(output_path, 'webp', quality=80)
            print(f"Converted: {filename} -> {name}.webp")
            
            # Optional: os.remove(file_path)
        except Exception as e:
            print(f"Error converting {filename}: {e}")

print("All done!")
